sclorg/betka
# Initial propose

[EDITED] Betka, a bot, is used for syncing upstream into downstream.
It is able to create a PR in src.fedoraproject.org in
dist-git repository based on upstream repository

How to review this PR? There are two ways:
1. Running test suite
- Create a file `secrets.env` with values like:
```bash
GIHUB_API_TOKEN=dummy
PAGURE_API_TOKEN=dummy
```
- Run `make test-in-container`

2. Run real sync:
- Get GITHUB_API_TOKEN from `https://github.com/settings/tokens` If it does not exist, then create it
- Get PAGURE_API_TOKEN from `https://src.fedoraproject.org/settings#nav-api-tab`. If it does not exist. then create it.
- Save both tokens into `secrets.env` file
- in a terminal run command `make run` and wait a minute till you don't see messages like:
```bash
betka_1      | [2020-04-02 11:57:55,097: DEBUG/MainProcess] ^-- substep ok
betka_1      | [2020-04-02 11:57:55,097: DEBUG/MainProcess] | Consumer: Starting event loop
betka_1      | [2020-04-02 11:57:55,097: DEBUG/MainProcess] | Worker: Hub.register Pool...
betka_1      | [2020-04-02 11:57:55,098: INFO/MainProcess] celery@docker-desktop ready.
betka_1      | [2020-04-02 11:57:55,098: DEBUG/MainProcess] basic.qos: prefetch_count->4
```
- in a second terminal run a command: `make send-master-sync`.

At the end, a new PR in `https://src.fedoraproject.org/container/s2i-core/pull-requests` should be generated.

For now, we are syncing `https://github.org/sclorg/s2i-base-container` into `src.fedoraproject.org/container/s2i-core` only.

Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>
